packages/story-editor/src/components/canvas/karma/canvasKeyboardNavigation.karma.js

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import { Fixture } from '../../../karma';
+import { useStory } from '../../../app/story';
+
+describe('Canvas - keyboard navigation', () => {
+  let fixture;
+
+  beforeEach(async () => {
+    fixture = new Fixture();
+
+    await fixture.render();
+    await fixture.collapseHelpCenter();
+  });
+
+  afterEach(() => {
+    fixture.restore();
+  });
+
+  it('should not focus the canvas while tabbing through the editor', async () => {
+    const focusContainer = fixture.screen.getByTestId('canvas-focus-container');
+
+    // start focused on media pane searchbar
+    await fixture.events.focus(fixture.editor.library.media.searchBar);
+
+    // tab until focus reaches the canvas container
+    let count = 0;
+    while (count < 50) {
```

Comment:
just curious, how'd you land on 50?

Reply:
Picked a number that felt large but not too large - but 50 is a very large overshoot. It only takes 4 tabs to get to the focus container area, so we could always bring this down. Just wanted to have an upper bound on this loop

maybe 10 or 15 is good :)